src/tests/sqlsmith/src/sql_gen/query.rs

```diff
@@ -34,7 +34,7 @@ impl<'a, R: Rng> SqlGenerator<'a, R> {
     /// Generates query expression and returns its
     /// query schema as well.
     pub(crate) fn gen_query(&mut self) -> (Query, Vec<Column>) {
-        if self.can_recurse() {
+        if self.rng.gen_bool(0.3) {
```

Comment:
Simple query can still recurse. Using `can_recurse` is misleading here. So just use a weight instead.